src/screens/TopicMemberScreen/index.js

```diff
@@ -48,35 +40,58 @@ const TopicMemberScreen = () => {
   const [profile] = React.useContext(Context).profile;
   const [isInitialLoading, setIsInitialLoading] = React.useState(true);
   const [headerHide, setHeaderHide] = React.useState(false);
-
+  const [searchHeight, setSearchHeight] = React.useState(0);
   const [searchText, setSearchText] = React.useState('');
   const [initalMember, setInitialMember] = React.useState([]);
   const [topicDataFollowedUsers, setTopicDataFollowedUsers] = React.useState([]);
   const [topicDataUnfollowedUsers, setTopicDataUnfollowedUsers] = React.useState([]);
   const [isFocus, setIsFocus] = React.useState(true);
   const [isFirstTimeOpen, setIsFirstTimeOpen] = React.useState(true);
   const cancelTokenRef = React.useRef(axios.CancelToken.source());
+  const scrollY = React.useRef(new Animated.Value(0)).current;
 
-  const interactionManagerRef = React.useRef(null);
-  const interactionManagerAnimatedRef = React.useRef(null);
+  const [isLoadingDiscovery, setIsLoadingDiscovery] = React.useState({
+    user: false
+  });
 
-  const opacityAnimationHeader = React.useRef(new Animated.Value(1)).current;
+  const pathCover = topicDetail?.cover_path || null;
+  const positionTop = Platform.OS === 'ios' ? top : 0;
 
-  const coverPath = topicDetail?.cover_path || null;
-  const topPosition = Platform.OS === 'ios' ? top : 0;
+  const navigationHeight = pathCover
+    ? dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT_COVER
+    : dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT;
+  const headerShowHeight =
+    navigationHeight + dimen.size.TOPIC_FEED_HEADER_HEIGHT + positionTop + searchHeight;
+  const headerHideHeight = dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT2 + positionTop + searchHeight;
 
-  const animatedHeight = React.useRef(
-    new Animated.Value(
-      (coverPath
-        ? dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT_COVER
-        : dimen.size.TOPIC_FEED_NAVIGATION_HEIGHT) +
-        dimen.size.TOPIC_FEED_HEADER_HEIGHT +
-        topPosition
-    )
-  ).current;
+  const animatedHeight = scrollY.interpolate({
+    inputRange: [0, headerHideHeight],
+    outputRange: [headerShowHeight, headerHideHeight],
+    extrapolate: 'clamp'
+  });
 
-  const [isLoadingDiscovery, setIsLoadingDiscovery] = React.useState({
-    user: false
+  const opacityAnimationHeader = scrollY.interpolate({
+    inputRange: [0, 100],
+    outputRange: [1, 0],
+    extrapolate: 'clamp'
+  });
+
+  React.useEffect(() => {
+    const listener = scrollY.addListener(({value}) => {
+      if (value >= 0 && value <= headerHideHeight) {
+        setHeaderHide(false);
+      } else if (value >= headerHideHeight) {
+        setHeaderHide(true);
+      }
+    });
+
+    return () => {
+      scrollY.removeListener(listener);
+    };
+  }, [scrollY]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Several new state variables have been introduced (`searchHeight`, `isFocus`, `isFirstTimeOpen`) and some existing ones have been removed (`interactionManagerRef`, `interactionManagerAnimatedRef`, `opacityAnimationHeader`). The `scrollY` variable has also been changed from a ref to a state variable. These changes could significantly alter the behavior of the component. Make sure all new state variables are properly initialized and updated, and that all removed state variables are not referenced elsewhere in the component.
```diff
-  const interactionManagerRef = React.useRef(null);
-  const interactionManagerAnimatedRef = React.useRef(null);
-  const opacityAnimationHeader = React.useRef(new Animated.Value(1)).current;
+  const [searchHeight, setSearchHeight] = React.useState(0);
+  const [isFocus, setIsFocus] = React.useState(true);
+  const [isFirstTimeOpen, setIsFirstTimeOpen] = React.useState(true);
+  const scrollY = React.useRef(new Animated.Value(0)).current;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->